src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            int[] arr = readFile(fromFileName);
+            String result = getResult(arr);
+            writeFile(toFileName, result);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private int[] readFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            int supplySum = 0;
+            int buySum = 0;
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] operation = line.split(",");
+                final String operationName = operation[0];
+                final String operationValue = operation[1];
+                if (operationName.equals(OPERATION_SUPPLY)) {
+                    supplySum += Integer.parseInt(operationValue);
```

Comment:
Here, you have duplication code. Feel free to do parsing before `if` checks

Reply:
Still exist.